service/kernel/verbose_mode.cpp

```diff
@@ -0,0 +1,222 @@
+{
+    end();
+    return *this;
+}
+
+void json::finalize()
+{
+    for (int i = 0; i < depth; ++i) write('}');
+    depth = 0;
+    write('\n');
+}
+
+void json::comma_if_needed()
+{
+    if (need_comma) write(',');
+}
+
+void json::begin()
+{
+    comma_if_needed();
+    need_comma = false;
+    write('{');
+    ++depth;
+}
+
+void json::end()
+{
+    --depth;
+    write('}');
+}
+
+void json::write_escape(const char * const str)
+{
+    write('"');
+    for (const char * c = str; *c; ++c)
+    {
+        switch (*c)
+        {
+        case '"': write("\\\""); break;
+        case '\\': write("\\\\"); break;
+        case '/': write("\\/"); break;
+        case '\b': write("\\b"); break;
+        case '\f': write("\\f"); break;
+        case '\n': write("\\n"); break;
+        case '\r': write("\\r"); break;
+        case '\t': write("\\t"); break;
+
+        default: write(*c); break;
+        }
+    }
+    write('"');
+}
+
+json::~json()
+{
+    finalize();
+}
+
+void json::write(const char * const str)
+{
+    if (0 > std::printf("%s", str)) std::terminate();
+}
+
+void json::write(const char c)
+{
+    if (0 > std::printf("%c", c)) std::terminate();
+}
+
+void json::write(const int i)
+{
+    if (0 > std::printf("%d", i)) std::terminate();
+}
+
+void json::write(const unsigned int u)
+{
+    if (0 > std::printf("%u", u)) std::terminate();
+}
+
+void json::write(const unsigned long u)
+{
+    if (0 > std::printf("%lu", u)) std::terminate();
+}
+
+void json::write(const long long int i)
+{
+    if (0 > std::printf("%lld", i)) std::terminate();
+}
+
+void json::write(const unsigned long long int u)
+{
+    if (0 > std::printf("%llu", u)) std::terminate();
+}
+
+void json::write(const double d)
+{
+    if (0 > std::printf("%f", d)) std::terminate();
+}
+
+// algorithms::kmeans::Parameter &
+void json_print(json & writer, const algorithms::kmeans::Parameter & val)
+{
+    writer.put("nClusters", val.nClusters).put("maxIterations", val.maxIterations).put("accuracyThreshold", val.accuracyThreshold);
+    writer.put("gamma", val.gamma).put("distanceType", val.distanceType).put("assignFlag", val.assignFlag);
+}
+
+// data_management::NumericTable &
+void json_print(json & writer, const data_management::NumericTable & val)
+{
+    writer.put("numberOfColumns", val.getNumberOfColumns()).put("numberOfRows", val.getNumberOfRows());
+    writer.put("dataLayout", val.getDataLayout()).put("dataMemoryStatus", val.getDataMemoryStatus());
+    // if verbose level 3 - show small part of array
+}
+
+void json_print(json & writer, ...) {}
+
+verbose_t::verbose_t()
+{
+    const char * const var = std::getenv("DAAL_VERBOSE");
+    if (var)
+    {
+        if ('1' == var[0] && '\0' == var[1]) verbose_t::level = 1;
+        if ('2' == var[0] && '\0' == var[1]) verbose_t::level = 2;
+    }
+    else
+    {
+        verbose_t::level = 0;
+    }
+}
+
+int verbose_t::level;
+
+static const ::daal::service::verbose_mode::verbose_t static_init_verbose_mode;
```

Comment:
Use safer approach for singleton, to avoid wrong level value in other singletons.